operator/v1/types_console.go

```diff
@@ -156,14 +156,15 @@ const (
 	CatalogTypeDisabled CatalogTypesState = "Disabled"
 )
 
-// DeveloperConsoleCatalogTypesState defines the state of the sub-catalog types.
-// +kubebuilder:validation:XValidation:rule="has(self.state) && self.state == 'Enabled' && has(self.disabled)",message="disabled is forbidden when state is Enabled."
-// +kubebuilder:validation:XValidation:rule="has(self.state) && self.state == 'Disabled' && has(self.enabled)",message="enabled is forbidden when state is Disabled."
+// DeveloperConsoleCatalogTypes defines the state of the sub-catalog types.
+// +kubebuilder:validation:XValidation:rule="self.state == 'Enabled' ? true : !has(self.enabled)",message="enabled is forbidden when state is not Enabled"
+// +kubebuilder:validation:XValidation:rule="self.state == 'Disabled' ? true : !has(self.disabled)",message="disabled is forbidden when state is not Disabled"
 // +union
-type DeveloperConsoleCatalogTypesState struct {
+type DeveloperConsoleCatalogTypes struct {
 	// state defines if a list of catalog types should be enabled or disabled.
 	// +unionDiscriminator
 	// +kubebuilder:validation:Enum:="Enabled";"Disabled";
+	// +kubebuilder:default:="Enabled"
```

Comment:
Something I forgot, can you add

// +default=Enabled


As well as this and regenerate the openapi definition, this is required for the generated apply client